### PR TITLE
build/configs/rtl8721csm: Using the single Make.defs file for cppload…

### PR DESCRIPTION
**Pull request by Tirtha-Samsung**
…able

### CODE REVIEW

File: build/configs/rtl8721csm/Make.defs [hunk 1]
```diff
@@ -61,9 +61,9 @@ LDSCRIPT = rlx8721d_img2_is.ld
 endif
 
 ifeq ($(CONFIG_LIBCXX_HAVE_LIBSUPCXX),y)
-LIBSUPXX = ${shell $(CC) $(CXXFLAGS) --print-file-name=libsupc++.a}
+LIBSUPXX = ${shell $(CC) $(CXXFLAGS) --print-file-name=libsupc++_nano.a}
 EXTRA_LIBPATHS += -L "${shell dirname "$(LIBSUPXX)"}"
-EXTRA_LIBS += -lsupc++
+EXTRA_LIBS += -lsupc++_nano
```
Comment (ewoodev) [inline build/configs/rtl8721csm/Make.defs:66]: other configs were using `-lsupc++`,
Could you please add commit comment why changed this?

File: build/configs/rtl8721csm/Make.defs [hunk 2]
```diff
@@ -161,10 +161,16 @@ CXXFLAGS = $(ARCHCXXFLAGS) $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS)
 
 ifeq ($(CONFIG_HAVE_CXX),y)
   CXXFLAGS += -std=c++11 -DCLOCK_MONOTONIC -D__TINYARA__
-  CXXFLAGS += -fexceptions -fcheck-new -frtti
+  CXXFLAGS += -fcheck-new -frtti
   CXXFLAGS += -pedantic -D_DEBUG -D_LIBCPP_BUILD_STATIC -ffunction-sections -fdata-sections -D__GLIBCXX__
 endif
 
+ifeq ($(CONFIG_HAVE_CXXINITIALIZE),y)
```
Comment (abhishek-samsung) [inline build/configs/rtl8721csm/Make.defs:168]: I think this should be app binary separation. For cpp in loadable, we need to provide -fno-exceptions flag. Using this config will be unclear.

File: build/configs/rtl8721csm/Make.defs (hunk 1 above)
Comment (abhishek-samsung) [inline build/configs/rtl8721csm/Make.defs:66]: For flat we can still use lsupc++, only for loadable we need to use nano as excpetions related changes are not yet supported for loadable

File: build/configs/rtl8721csm/Make.defs (hunk 2 above)
Comment (kishore-sn) [inline build/configs/rtl8721csm/Make.defs:168]: There is a config CONFIG_LIBCXX_EXCEPTION. Let us set this flag based on this config and disable this config in loadable.

File: build/configs/rtl8721csm/Make.defs [hunk 3]
```diff
@@ -139,7 +139,7 @@ endif
 ARCHCXXFLAGS = -fno-builtin -fexceptions -mcpu=cortex-m33 -mfpu=fpv5-sp-d16
 
 ifeq ($(CONFIG_HAVE_CXX),y)
-  ARCHCXXFLAGS = -fno-builtin $(ARCHCPUFLAGS)
+#  ARCHCXXFLAGS = -fno-builtin $(ARCHCPUFLAGS)
```
Comment (kishore-sn) [inline build/configs/rtl8721csm/Make.defs:142]: Why are we commenting this? Do we not need cpu flags for building cxx?

File: build/configs/rtl8721csm/Make.defs (hunk 1 above)
Comment (kishore-sn) [inline build/configs/rtl8721csm/Make.defs:66]: If this change is only related to exceptions, then let us put it under CONFIG_LIBCXX_EXCEPTION.

Comment (abhishek-samsung) [reply build/configs/rtl8721csm/Make.defs:66]: This sounds good. @Tirtha-Samsung please verify the booting of the board on cxxtest config too with the changes. I haven't tested libsupc++_nano with flat build before.

Comment (Tirtha-Samsung) [reply build/configs/rtl8721csm/Make.defs:66]: I will verify it

**Review by sunghan-chang — COMMENTED** [review]
We don't allow empty commit description. Please write what / why of the issue.

**Review by sunghan-chang — COMMENTED** [review]
We don't allow empty commit description. Please write what / why of the issue.

**Comment by thapav** [comment]
Please update commit message to something more suitable, may be something like
Merge cpp loadable configuration Make.defs with common Make.defs for tp1x

File: build/configs/rtl8721csm/Make.defs (hunk 1 above)
Comment (Tirtha-Samsung) [reply build/configs/rtl8721csm/Make.defs:66]: lsupc++_nano doesnot execute the exception libraries and for loadable we donot need exeption ig

**Comment by sunghan-chang** [comment]
> We don't allow empty commit description. Please write what / why of the issue. I don't know why you modify this.

@Tirtha-Samsung Commit description still does not explain the change. Why do you change `-lsupc++` to `-lsupc++_nano`? Why do you add `CONFIG_LIBCXX_EXCEPTION` checking?
I think the issue is
single Make.defs should support entire defconfig including cpp on loadable. But current one does not. TizenRT does not support C++ exception on loadable so that we should use `-lsupc++_nano` and should exclude -fexception option by CONFIG_LIBCXX_EXCEPTION control.

**Review by sunghan-chang — APPROVED** [review]

File: build/configs/rtl8721csm/Make.defs (hunk 1 above)
Comment (sunghan-chang) [reply build/configs/rtl8721csm/Make.defs:71]: > This sounds good. @Tirtha-Samsung please verify the booting of the board on cxxtest config too with the changes. I haven't tested libsupc++_nano with flat build before.

@Tirtha-Samsung Is this verified on the flat?

Comment (Tirtha-Samsung) [reply build/configs/rtl8721csm/Make.defs:71]: Yes I verified cxxtest->helloxx, it is booting fine

**Pull request merged**